src/Text.tsx

```diff
@@ -1,13 +1,28 @@
+import React from 'react'
 import styled from 'styled-components'
 import {COMMON, SystemCommonProps, SystemTypographyProps, TYPOGRAPHY} from './constants'
 import sx, {SxProp} from './sx'
-import {ComponentProps} from './utils/types'
 
-const Text = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
+const RestrictedAs = {
```

Comment:
Is there a reason you're using PascalCase for this object?

Reply:
Nope, I'll swap this out for camel.